File: db-python-ex/example/DB_01_ConnectPostgreSQL.py
import psycopg2

# PostgreSQL 데이터베이스 연결
connection = psycopg2.connect(
    host="localhost",
    port="5433",
    user="admin",
    password="postgres",
    database="postgres"
)

# 커넥션 상태 확인
if connection:
    print("PostgreSQL DB 연결 성공")
else:
    print("PostgreSQL DB 연결 실패")
    exit(100)

# 접속된 데이터베이스에서 쿼리 실행
cursor = connection.cursor()
cursor.execute("SELECT * FROM category LIMIT 100")
result = cursor.fetchall()

# 쿼리 결과 출력
for row in result:
    print(row)

# 연결 종료
cursor.close()
connection.close()

File: db-perl-ex/src/DB_01_ConnectPostgreSQL.pl
use DBI;

# PostgreSQL 데이터베이스 접속 정보
my $host = "localhost";
my $port = "5433";
my $user = "admin";
my $password = "postgres";
my $database = "postgres";

# PostgreSQL 데이터베이스에 접속
my $dsn = "DBI:Pg:database=$database;host=$host;port=$port";
my $dbh = DBI->connect($dsn, $user, $password) or die "Cannot connect to the database: $DBI::errstr";

# SELECT 쿼리 실행
my $sql = "SELECT * FROM category LIMIT 20";
my $sth = $dbh->prepare($sql);
$sth->execute();

# 결과 출력
print "result 1 assign column\n";
while (my $row = $sth->fetchrow_hashref) {
    print "Column1: " . $row->{category_major_cd} . ", Column2: " . $row->{category_major_name} . "\n";
}

$sth->execute();
print "result 2 array\n";
while (my @row = $sth->fetchrow_array) {
    print join(",", @row) . "\n";
}

# 연결 종료
$sth->finish;
$dbh->disconnect;